### ISSUE TITLE
Enable addon previews (screenshots) to be added/deleted via the addon api

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#18236]
split from #18036

We want to be able to add and remove (and edit) the addon previews/screenshots via the api, like we can in devhub.  Because there can be many we'll likley want a sub-endpoint under addon dedicated to creating, listing, deleting, etc, the previews.

**Comment by ioanarusiczki** [comment]
@eviljeff Could you add some details about this? 

 I did not find instructions in the docs about previews so I tried attaching a .jpg to "preview" as form-data (same way as for icon) but it does not seem to work.

**Comment by ioanarusiczki** [comment]
@eviljeff found something here https://github.com/mozilla/addons-server/pull/19091/files#diff-c2b741fadc1162d0339214429b8efe1e6827c92fff032fa5d7068af81b9bbd71

**Comment by eviljeff** [comment]
@ioanarusiczki because there are multiple screenshots there's a separate (sub) endpoint:
https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#preview-create

**Comment by ioanarusiczki** [comment]
@eviljeff 
I verified POST, PATCH and DELETE on -dev for extensions and it looks good:
- I could add  previews, edit and delete them.
- I get 400 responses (as expected) for a couple of situations:
"Images cannot be animated."
"Images must be smaller than 4MB"
"Images must be either PNG or JPG."
"Upload a valid image. The file you uploaded was either not an image or a corrupted image."

 I ran into a situation: after I tried PATCH and DELETE on a theme id -> I get no response, Status and Time are blank. 
POST  create preview is returning status created but sending a preview for a theme has no impact in dev hub. 
I replaced the id of the theme with an extension after this but I still get no response for PATCH and DELETE  so I made new requests (into a new tab) which will work as expected. 
Maybe the attempt to send a preview for a theme should throw an error ?

PATCH/DELETE after using a theme id/guid:

![does not respond](https://user-images.githubusercontent.com/33448286/163136659-977c2f91-e8d6-4bf3-8bb9-97c802389e67.png)
